source/tests/system/nitclk_driver_api_tests.cpp

```diff
@@ -0,0 +1,268 @@
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    tclk::GetAttributeViStringResponse response;
+
+    ::grpc::Status status = GetTClkStub()->GetAttributeViString(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+    return response.value();
+  }
+  
+  void set_real64_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id, const ViReal64 attribute_value)
+  {
+    ::grpc::ClientContext context;
+    tclk::SetAttributeViReal64Request request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    request.set_value(attribute_value);
+    tclk::SetAttributeViReal64Response response;
+
+    ::grpc::Status status = GetTClkStub()->SetAttributeViReal64(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+  }
+
+  void set_session_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id, const int session_id)
+  {
+    ::grpc::ClientContext context;
+    tclk::SetAttributeViSessionRequest request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    request.mutable_value()->set_id(session_id);
+    tclk::SetAttributeViSessionResponse response;
+
+    ::grpc::Status status = GetTClkStub()->SetAttributeViSession(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+  }
+
+  void set_string_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id, const ViString attribute_value)
+  {
+    ::grpc::ClientContext context;
+    tclk::SetAttributeViStringRequest request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    request.set_value(attribute_value);
+    tclk::SetAttributeViStringResponse response;
+
+    ::grpc::Status status = GetTClkStub()->SetAttributeViString(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+  }
+
+ private:
+  std::shared_ptr<::grpc::Channel> channel_;
+  std::unique_ptr<::nidevice_grpc::Session> scope_session_;
+  std::unique_ptr<scope::NiScope::Stub> niscope_stub_;
+  std::unique_ptr<tclk::NiTClk::Stub> nitclk_stub_;
+  std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+  std::unique_ptr<scope::NiScopeLibrary> niscope_library_;
+  std::unique_ptr<scope::NiScopeService> niscope_service_;
+  std::unique_ptr<tclk::NiTClkLibrary> nitclk_library_;
+  std::unique_ptr<tclk::NiTClkService> nitclk_service_;
+  std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiTClkDriverApiTest, ConfigureForHomogeneousTriggers_ReturnsStatusAsSuccessful)
```

Comment:
@doshirohan , for the test name, if you have better suggestions, please let me know.

Reply:
This name seems fine.